### ISSUE TITLE
.pyw files aren't reported properly

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#261]
[Reported by Timothy Wasserman]

(OS = Win7, Coverage version = 3.6, Python = 2.7.3, wxPython = 2.8.12.1)

I’m working on a wxPython app in windows, and I’ve given some of my source files the .pyw extension so that they open without a console window.

From the command prompt, when I do ‘coverage run test_mop.py’ it runs without complaint, but when I subsequently try ‘coverage report’ it gives me a NoSource error like this:
```
C:\Users\twasserm\Desktop\workspace\FDS\test>coverage report

Name                                                  Stmts   Miss  Cover
-------------------------------------------------------------------------
C:\Users\twasserm\Desktop\workspace\FDS\src\fdsutil      49     25    49%
C:\Users\twasserm\Desktop\workspace\FDS\src\mop   NoSource: No source for code: 'C:\Users\twasserm\Desktop\workspace\FDS\src\mop.py'
test_mop                                                 44      0   100%
-------------------------------------------------------------------------
TOTAL                                                    93     25    73%
```

The mop.py source file that coverage report is looking for is  actually mop.pyw. fdsutil has a standard .py extension and is correctly processed. When I make a copy of mop.pyw as mop.py, ‘coverage report’ picks it up immediately (without having to do coverage run again):

```
C:\Users\twasserm\Desktop\workspace\FDS\test>coverage report
Name                                                  Stmts   Miss  Cover
-------------------------------------------------------------------------
C:\Users\twasserm\Desktop\workspace\FDS\src\fdsutil      49     25    49%
C:\Users\twasserm\Desktop\workspace\FDS\src\mop         121     73    40%
test_mop                                                 44      0   100%
-------------------------------------------------------------------------
TOTAL                                                   214     98    54%
```

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/261

**Comment by nedbat** [comment]
Commit [783923fd9ccccbb (bb)](https://bitbucket.org/ned/coveragepy/commits/783923fd9ccccbb)  helped some with this.

**Comment by nedbat** [comment]
Fixed in [4417e335f3e4 (bb)](https://bitbucket.org/ned/coveragepy/commits/4417e335f3e4)

**Comment by nedbat** [comment]
*Original comment by* **Okko Willeboordse (Bitbucket: [owillebo](https://bitbucket.org/owillebo), GitHub: [owillebo](https://github.com/owillebo))**

----------------------------------------

==============
Update;
Now using version 4.0.3 and this version solves problem described below.
==============



I have two files;

test.py;
```
#!python
import coverage
cov = coverage.coverage(source = ['target'])
cov.start()
import target
cov.stop()
print cov.html_report()
```

and target.py(w);
```
#!python
def foo():
  pass
```

When I run test.py with target.py it outputs, as expected;

50.0

But when I rename target.py in to target.pyw, test.py outputs;
```
#!python
Coverage.py warning: Module target has no Python source.
Coverage.py warning: No data was collected.
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    print cov.html_report()
  File "c:\Python27\lib\site-packages\coverage\control.py", line 662, in html_report
    return reporter.report(morfs)
  File "c:\Python27\lib\site-packages\coverage\html.py", line 113, in report
    self.report_files(self.html_file, morfs, self.config.html_dir)
  File "c:\Python27\lib\site-packages\coverage\report.py", line 76, in report_files
    raise CoverageException("No data to report.")
coverage.misc.CoverageException: No data to report.
```

How do I run coverage programmatically with .pyw files?

Note that Python imports test(.pyw) just the same.
